### PR TITLE
TEC-4765 create linkedin organisation operations

### PR DESCRIPTION
**Pull request by diepjy**
### Description of Changes

Added Organization operations in a wrapper class `OrganizationConnection` for only the minimal operations needed. 
Happy to discuss if any have been missed out

### Documentation

https://docs.microsoft.com/en-us/linkedin/marketing/integrations/community-management/organizations

### Risks & Impacts

This will be the main entry point with interacting with the LinkedIn Organization API

### Testing

Ran each method to ensure parameters and URL are formatted correctly.
Do not have direct access to V2 so will need to be retested once we have been granted access.

### CODE REVIEW

File: src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+   * E.g. https://api.linkedin.com/v2/organizationalEntityAcls?q=roleAssignee
+   * @param role Limit results to specific roles, such as ADMINISTRATOR.
+   * @param state Limit results to specific role states, such as APPROVED.
+   * @param projection Field projection
+   * @return List of access controls for a given role and state for the member
+   */
+  public List<AccessControl> fetchMemeberOrganizationAccessControl(String role, String state,
+      Parameter projection) {
+    Parameter queryParam = Parameter.with(QUERY_KEY, ROLE_ASSIGNEE_VALUE);
+    Parameter roleParam = Parameter.with(ROLE_KEY, role);
```
Comment (am6010) [inline src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java]: If role is null we will get an exception. Role and state are not required fields and if null maybe we should not include them as you do below

File: src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+  }
+
+  private void addTimeIntervalToQueryParameters(TimeInterval timeInterval,
+      List<Parameter> parameters) {
+    if (timeInterval != null) {
+      if (!StringUtils.isBlank(timeInterval.getTimeGranularityType())) {
+        parameters.add(Parameter.with("timeIntervals.timeRange",
+            timeInterval.getTimeGranularityType()));
+      }
+      if (timeInterval.getTimeRange() != null && timeInterval.getTimeRange().getStart() != null
```
Comment (am6010) [inline src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java]: Start or end can be null but not at the same time. Do we want to include this case?

File: src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+    ValidationUtils.verifyParameterPresence("share", shareURN);
+    if (!URNEntityType.SHARE.equals(shareURN.getURNEntityType())) {
+      throw new IllegalArgumentException("The URN should be type SHARE");
+    }
+  }
+
+  private void addTimeIntervalToQueryParameters(TimeInterval timeInterval,
+      List<Parameter> parameters) {
+    if (timeInterval != null) {
+      if (!StringUtils.isBlank(timeInterval.getTimeGranularityType())) {
```
Comment (am6010) [inline src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java]: `isNotBlank` ?

File: src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+   * @param timeInterval the time interval for time-bounded statistics
+   * @param shareURNs specific share URNs
+   * @return List of organization share statistics
+   */
+  public List<Statistics> retrieveOrganizationShareStatistics(URN organizationURN,
+      TimeInterval timeInterval, List<URN> shareURNs) {
+    validateOrganizationURN("organizationURN", organizationURN);
+
+    List<Parameter> parameters = new ArrayList<>();
+    parameters.add(Parameter.with(QUERY_KEY, ORGANIZATION_VALUE));
```
Comment (am6010) [inline src/main/java/com/echobox/api/linkedin/connection/OrganizationConnection.java]: Should this one be `ORGANIZATION_VALUE_ENTITY`?

File: src/main/java/com/echobox/api/linkedin/connection/ConnectionBase.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * The connection base class
+ * @author joanna
+ *
+ */
+public abstract class ConnectionBase {
+  
+  protected LinkedInClient linkedinClient;
+  
+  protected ConnectionBase(LinkedInClient linkedinClient) {
+    this.linkedinClient = linkedinClient;
```
[the remark this answers is not in the document]
Comment (diepjy) [reply src/main/java/com/echobox/api/linkedin/connection/ConnectionBase.java]: Yeah, can add for extra safety

File: src/main/java/com/echobox/api/linkedin/connection/ConnectionBase.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    Iterator<List<T>> iterator = connection.iterator();
+    while (iterator.hasNext()) {
+      resultList.addAll(iterator.next());
+    }
+
+    return resultList;
+  }
+  
+  protected void validateURN(URNEntityType urnEntityType, URN urn) {
```
[the remark this answers is not in the document]
Comment (diepjy) [reply src/main/java/com/echobox/api/linkedin/connection/ConnectionBase.java]: Ah was going to use it in the validation methods for code dupe

**Pull request merged**